### PR TITLE
Fix const iterator issue1437

### PR DESCRIPTION
**Pull request by kerautret**
# PR Description

 See issue #1437 

# Checklist

- [x] Unit-test of your feature with [Catch](http://dgtal.org/doc/stable/moduleCatch.html).
- [x] Doxygen documentation of the code completed (classes, methods, types, members...)
- [x] Documentation module page added or updated.
- [x] New entry in the [ChangeLog.md](https://github.com/DGtal-team/DGtal/blob/master/ChangeLog.md) added.
- [x] No warning raised in Debug ```cmake``` mode (otherwise, Travis C.I. will fail).
- [x] All continuous integration tests pass (Travis & appveyor)

### CODE REVIEW

File: ChangeLog.md [hunk 1]
```diff
@@ -135,7 +135,10 @@
   (David Coeurjolly, [#1446](https://github.com/DGtal-team/DGtal/pull/1446))
   - Removes the check for CPP11 when building WITH_ITK
   (Pablo Hernandez-Cerdan, [#1453](https://github.com/DGtal-team/DGtal/pull/1453))
-
+  - Fix apple clang  compilation issue with a workaround to the
+    ConstIteratorAdapter that do not satisfy the _is_forward concept of the STL:
```
Comment (dcoeurjo) [inline ChangeLog.md:139]:
```suggestion
    ConstIteratorAdapter class that does not satisfy the _is_forward concept of the STL:
```

**Comment by dcoeurjo** [comment]
thx for the PR

File: ChangeLog.md (hunk 1 above)
Comment (kerautret) [reply ChangeLog.md]: thanks, I miss the commit suggestion since my emacs was open on it but that looks nice!

**Comment by dcoeurjo** [comment]
All good ;)

**Pull request merged**